src/ui/public/vislib/components/zero_injection/inject_zeros.js

```diff
@@ -29,12 +29,12 @@ export default function ZeroInjectionUtilService(Private) {
     }
   }
 
-  return function (obj) {
+  return function (obj, sortBuckets = false) {
     if (!_.isObject(obj) || !obj.rows && !obj.columns && !obj.series) {
       throw new TypeError('ZeroInjectionUtilService expects an object with a series, rows, or columns key');
     }
 
-    const keys = orderXValues(obj);
+    const keys = orderXValues(obj, sortBuckets);
```

Comment:
I think renaming this to `orderBucketsBySum` everywhere will make it a little clearer, since technically the buckets are going to be sorted anyway, right? They'll just be sorted by their index by default.